src/qibo/tomography_RGD/qibo_states.py

```diff
@@ -0,0 +1,257 @@
+
+    @staticmethod
+    def make_measurement_circuit_name(circuit_name, label):
+        """
+        Measurement circuit naming convention
+        """
+        name = "{}-{}".format(circuit_name, label)
+        return name
+
+    @staticmethod
+    def sort_sample(sample) -> dict:
+        """
+        Sort raw measurements into count dictionary
+        """
+        sorted_sample = {}
+        for shot in sample:
+            s = "".join(map(str, shot))
+            if s in sorted_sample:
+                sorted_sample[s] += 1
+            else:
+                sorted_sample[s] = 1
+        return sorted_sample
+
+    def execute_measurement_circuits(
+        self,
+        labels,
+        backend="numpy",
+        num_shots=100,
+        # num_shots = 10000,
+        label_format="little_endian",
+    ):
+        """
+        Executes measurement circuits
+        - labels: string of Pauli matrices (e.g. XYZXX)
+        - backend: 'numpy', 'qibojit', 'pytorch', 'tensorflow' prvided by qibo
+        - num_shots: number of shots measurement is taken to get empirical frequency through counts
+        """
+        if self.measurement_circuit_names == []:
+            self.create_measurement_circuits(labels, label_format)
+
+        circuit_names = self.measurement_circuit_names
+
+        qibo.set_backend(backend)
+
+        data_dict_list = []
+        for i, label in enumerate(labels):
+            result = self.measurement_circuits[i].execute(nshots=num_shots)
+            result.samples(binary=True, registers=False)
+            count_dict = self.sort_sample(result.to_dict()["samples"])
+
+            measurement_circuit_name = self.make_measurement_circuit_name(
+                self.circuit_name, label
+            )
+            data_dict = {
+                "measurement_circuit_name": measurement_circuit_name,
+                "circuit_name": self.circuit_name,
+                "label": label,
+                "count_dict": count_dict,
+                "backend": "qibo: " + backend,
+                "num_shots": num_shots,
+            }
+            data_dict_list.append(data_dict)
+        return data_dict_list
+
+
+class GHZState(State):
+    """
+    Constructor for GHZState class
+    """
+
+    def __init__(self, n):
+        State.__init__(self, n)
+        self.circuit_name = "GHZ"
+
+    def create_circuit(self):
+        circuit = qibo.Circuit(self.n)
+
+        circuit.add(gates.H(0))
+        for i in range(1, self.n):
+            circuit.add(gates.CNOT(0, i))
+
+        self.circuit = circuit
+
+
+class HadamardState(State):
+    """
+    Constructor for HadamardState class
+    """
+
+    def __init__(self, n):
+        State.__init__(self, n)
+        self.circuit_name = "Hadamard"
+
+    def create_circuit(self):
+        circuit = qibo.Circuit(self.n)
+
+        for i in range(self.n):
+            circuit.add(gates.H(i))
+
+        self.circuit = circuit

```

Comment:
Since this is just a layer of Hadamards, this is a Hadamard transform of the initial state. A more concise way to implement this would be to modify the following function: https://github.com/qiboteam/qibo/blob/958b11bc38b800f217ad9c97b950635896935fd4/src/qibo/quantum_info/utils.py#L118
For instance, if the input is a circuit, then a layer of Hadamard is added. That would perform the Hadamard of whatever state one wants, including the zero state in your case.